src/bullets/utils/market_open_utils.py

```diff
@@ -0,0 +1,26 @@
+from datetime import datetime, timedelta, date
+from bullets.data_source.data_source_interface import Resolution
+from bullets.utils.holiday_date_util import us_holiday_list
+
+def is_market_open(date: datetime, resolution: Resolution=Resolution.DAILY) -> bool:
+        if date.weekday() >= 5:
+            return False
+
+        if resolution != Resolution.DAILY:
+            if date.hour < 9 or date.hour > 16:
+                return False
+            elif date.hour == 16 and date.minute > 0:
+                return False
+            elif date.hour == 9 and date.minute < 30:
+                return False
+
+        return date not in us_holiday_list(date.year)
+
+def get_date_x_days_away(x):
```

Comment:
x is not a valid name for a parameter
x should have a type specified
get_date_x_days_away should have a return type specified
the function has no documentation